### PR TITLE
Ign/fill implicit padding

### PR DESCRIPTION
**Pull request by ign-msati**
### Ticket
Link to Github Issue : #31983

### Problem description
Padding handling in fused and reduction operations in tt-metal / ttnn is incorrect when non-zero padding values are used.

This is due to the behavior of the ZEROACC instruction, which sets zero flags instead of clearing Dest register data. While the FPU and Packer honor these flags, the SFPU does not, leading to stale or undefined values being used in padded regions.

As a result, operations that involve the SFPU can produce incorrect results, since padding is not consistently treated as zero during computation. This leads to unreliable behavior for users, where valid inputs that are not tile-aligned may fail or produce incorrect results due to missing internal padding handling.

### What's changed

- Added implicit padding using a constant value TEST_PADDING_VALUE = -42 for TTNN input tensors

- Extended test coverage to include non-tiled tensor shapes (e.g., [1, 1, 15, 23]) to validate behavior under irregular dimensions

- Indentified issues while enabling non-zero padding:
    - #42135
    - #42148
- Temporarily skipped affected test cases with references to the corresponding issues

### CI Status
_Auto-generated on every push. Badges update live. Click a pipeline name or badge to filter by this branch._

| Pipeline | Status | Latest Run |
|---|:---:|---|
| [Sanity tests](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:ign/fill-implicit-padding) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:ign/fill-implicit-padding) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/sanity-tests.yaml?query=branch:ign/fill-implicit-padding) |
| [Blackhole post-commit](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:ign/fill-implicit-padding) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:ign/fill-implicit-padding) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/blackhole-post-commit.yaml?query=branch:ign/fill-implicit-padding) |
| [L2 Nightly](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ign/fill-implicit-padding) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ign/fill-implicit-padding) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/tt-metal-l2-nightly.yaml?query=branch:ign/fill-implicit-padding) |
| [(Single) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:ign/fill-implicit-padding) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:ign/fill-implicit-padding) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select.yaml?query=branch:ign/fill-implicit-padding) |
| [(T3K) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:ign/fill-implicit-padding) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:ign/fill-implicit-padding) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-t3k.yaml?query=branch:ign/fill-implicit-padding) |
| [(Galaxy) Choose your pipeline](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:ign/fill-implicit-padding) | [](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:ign/fill-implicit-padding) | [Launch ↗](https://github.com/tenstorrent/tt-metal/actions/workflows/pipeline-select-galaxy.yaml?query=branch:ign/fill-implicit-padding) |

### CODE REVIEW

File: tests/ttnn/unit_tests/operations/reduce/test_cumsum.py [hunk 1]
```diff
@@ -166,6 +170,7 @@ def test_cumsum_backward(size, dim, dtypes, device):
     # which are not handled yet
     torch_input_tensor = torch.randint(-2, 3, size=size, dtype=torch_dtype, requires_grad=True)
     input_tensor = ttnn.from_torch(torch_input_tensor, device=device, layout=ttnn.Layout.TILE)
+    input_tensor = ttnn.fill_implicit_tile_padding(input_tensor, TEST_PADDING_VALUE)
```
[the remark this answers is not in the document]
Comment (fplavecTT) [reply tests/ttnn/unit_tests/operations/reduce/test_cumsum.py]: I believe Copilot's suggestion was to remove both assignments to `input_tensor`, but only one was removed.

**Review by bbradelTT — APPROVED** [review]

File: tests/ttnn/unit_tests/operations/reduce/test_reduction.py [hunk 2]
```diff
@@ -733,6 +720,7 @@ def run_maxpool(device, input_shape, kernel_size, stride, padding, dilation):
     input_tensor = torch.permute(torch_input, (0, 2, 3, 1))
     input_tensor = torch.reshape(input_tensor, (1, 1, -1, in_c))
     input_tensor = ttnn.from_torch(input_tensor, layout=ttnn.ROW_MAJOR_LAYOUT, device=device)
+    input_tensor = ttnn.fill_implicit_tile_padding(input_tensor, TEST_PADDING_VALUE)
```
Comment (fplavecTT) [inline tests/ttnn/unit_tests/operations/reduce/test_reduction.py]: Since this test only uses row major layout, does padding do anything at this point?

File: tests/ttnn/unit_tests/operations/reduce/test_reduction_min.py [hunk 3]
```diff
@@ -12,6 +12,8 @@
 from tests.ttnn.utils_for_testing import assert_numeric_metrics
 from models.common.utility_functions import torch_random
 
+TEST_PADDING_VALUE = 42
```
Comment (fplavecTT) [inline tests/ttnn/unit_tests/operations/reduce/test_reduction_min.py]: This is not an effective padding value to test MIN since it's in the range of actual data (-100 to 100).
We should use a value smaller than any other in the range.

File: tests/ttnn/unit_tests/operations/reduce/test_cumsum.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ign-yaswanth) [reply tests/ttnn/unit_tests/operations/reduce/test_cumsum.py]: Fixed in the latest commit.

File: tests/ttnn/unit_tests/operations/reduce/test_reduction.py (hunk 2 above)
Comment (ign-yaswanth) [reply tests/ttnn/unit_tests/operations/reduce/test_reduction.py]: Fixed in the latest commit.

File: tests/ttnn/unit_tests/operations/reduce/test_reduction_min.py (hunk 3 above)
Comment (ign-yaswanth) [reply tests/ttnn/unit_tests/operations/reduce/test_reduction_min.py]: Fixed in the latest commit.

**Review by iwroszTT — APPROVED** [review]
